src/librustc_lint/context.rs

```diff
@@ -275,6 +284,23 @@ impl LintStore {
         }
     }
 
+    /// Check if the lint is gated on a feature that is not enabled.
+    pub fn check_gated_lint(&self, sess: &Session, lint_id: LintId, span: Span) {
+        if let Some((lint_name, feat)) = self
+            .gated_lints
+            .get(&lint_id)
+            .filter(|(_, feat)| !sess.features_untracked().enabled(*feat))
+        {
```

Comment:
Here, I think you would write something like:

```suggestion
        if let Some(feature) = lint_id.lint.feature_gate {
            if !sess.features_untracked().enabled(*feature) {
```